clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -94,6 +96,10 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
     )
     dataframe = dataframe[filename.isin(file_mod_list)]
     filename = filename[filename.isin(file_mod_list)]
+    if dataframe.empty:
+        raise ValueError(
```

Comment:
Could you add a unit test for `read_imaging_data` (just covering this corner case) ?
I suppose this error should be raised if you call `read_imaging_data` with an empty folder for example.

Reply:
Yup ! Looking into it, I wonder if it'll work though or how thorough we should be with this unit test. 
Should we just check that it errors right, or should we check that the data generated with a fake image is right as well ? I'm asking because it seems to imply generating complex fake data: fake .nii.gz inside zip files

Reply:
Even though it doesn't please me so much, I'd personally go all the way and generate it. WDYT ?

Reply:
Just check that the error is raised when providing no imaging data (basically only test what this PR is implementing). We'll work on improving the tests later to cover the rest of the function. Does this make sense?

Reply:
Alright ! This is much simpler 😄

Reply:
> Even though it doesn't please me so much, I'd personally go all the way and generate it. WDYT ?

If you are interested in this, why not! 
But I was only suggesting to add a non regression test for this fix.